packages/live-share-media/src/VolumeManager.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the Microsoft Live Share SDK License.
+ */
+
+import { TimeInterval } from '@microsoft/live-share';
+import { IMediaPlayer } from './IMediaPlayer';
+
+export enum LevelType { fixed, percentage }
+
+/**
+ * Smooth audio level changes when selectedVolume is modified, or if volume limiting has started/ended.
+ */
+export class VolumeManager {
+    private readonly _player: IMediaPlayer;
+    private readonly _volumeChangeDuration = new TimeInterval(500);
+
+    // defaults to player volume
+    private _selectedVolume = 0.0;
+    private _limited = false;
+    private _level = 0.1
+    private _levelType: LevelType = LevelType.fixed;
+    private _startTime = 0;
+    private _startVolume = 0;
+    private _running = false;
+
+    constructor(player: IMediaPlayer) {
+        this._player = player;
+        this._selectedVolume = this._player.volume;
+    }
+
+    /**
+     * The selected volume.
+     *
+     * @remarks
+     * Expressed as a value between 0.0 and 1.0. The default value is 1.0.
+     * Can be used for things like volume sliders.
+     */
+    public get selectedVolume(): number {
```

Comment:
>selectedVolume [](http://example.com/codeflow?start=15&length=14)

This should just be "volume" to match the property name on the player.

Reply:
There could be some confusion there, so i added another comment after renaming
`Does not return the current volume if limiting is enabled, but the desired volume.`

Reply:
Sounds good